spec/haskell/src/SEL4/Kernel/VSpace/ARM.lhs

```diff
@@ -1506,13 +1503,7 @@ notion of the largest permitted object size, and checks it appropriately.
 >         InvokeASIDPool oper -> do
 >             performASIDPoolInvocation oper
 >             return []
-#ifdef CONFIG_ARM_HYPERVISOR_SUPPORT
->         InvokeVCPU _ -> fail "performARMMMUInvocation: not an MMU invocation"
-#endif
-#ifdef CONFIG_ARM_SMMU
->         InvokeIOSpace _ -> fail "performARMMMUInvocation: not an MMU invocation"
->         InvokeIOPageTable _ -> fail "performARMMMUInvocation: not an MMU invocation"
-#endif
+>         _ -> fail "performARMMMUInvocation: not an MMU invocation"
```

Comment:
Isn't removing the SMMU/IOMMU parts and inserting a default case a bit much here? Same comment as for AARCH64, this will not catch future expansion in MMU invocations, and having a few extra lines is not such a high price for it.

Reply:
I found the repetition annoying, and it's going to get worse. It's not just the definition here where you have the repetition, you will also get repetition in `case_convs`, which is especially annoying because there you have to enumerate the parameters of those invocations as well. Unless we're doing SMMU, we're much more likely to add non-MMU invocations, so I think this is guarding the wrong case.